src/main/java/com/softserveinc/ita/pageobjects/ComparisonPage.java

```diff
@@ -28,6 +27,8 @@ public ComparisonPage removeProductFromCompare() {
         $("rz-trash-icon button")
                 .should(appear)
                 .click();
+        // need to wait because of rozetka's bug after deleting product
+        sleep(400);
```

Comment:
it's sad, but you need to avoid sleeps

try to add should with condition and timeout

$("rz-trash-icon button").shouldNotBe(visible, ofSeconds(3));
or should(not(visible), ofSeconds(3));
or shouldNotHave(cssClass or attribute)

try to think, what changes in the HTML, attribute, or class changing after clock on trash icon